spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/Column.java

```diff
@@ -294,9 +333,14 @@ String getPrefix() {
 	 */
 	static class AliasedColumn extends Column implements Aliased {
 
-		private final String alias;
+		private final SqlIdentifier alias;
 
 		private AliasedColumn(String name, Table table, String alias) {
+			super(name, table);
+			this.alias = SqlIdentifier.unquoted(alias);
```

Comment:
We're preserving the current functionality. Our quoting can be enabled or disabled on `PersistentEntity` level which is a different abstraction level. Since we're creating our SQL AST objects from these types, it's fair to not put the always quoting burden on every user of the SQL AST. You can always opt in with using `SqlIdentifier` on the consumer side if you're willing to control how identifiers are rendered.